### PR TITLE
Fix for ScalaJS tests issue #601

### PR DESCRIPTION
**Pull request by javaduke**

### CODE REVIEW

**Comment by javaduke** [comment]
This addresses issue #601

**Comment by javaduke** [comment]
> @javaduke this seems to have broken some of the existing tests. Do you think you could help investigate the travis failures?

I fixed the issue but now travis fails with some weird error, not sure how to fix it...

**Comment by lihaoyi** [comment]
@javaduke can you try rebasing and squashing your commits on top of the latest master? This is a travis bug and squashing normally makes it go away

**Comment by javaduke** [comment]
Yay, that did the trick, Travis works now. The Appveyor still fails though, do you know how to fix it?

File: scalajslib/worker/0.6/src/ScalaJSWorkerImpl.scala [hunk 1]
```diff
@@ -3,29 +3,33 @@ package scalajslib
...
 import org.scalajs.core.tools.jsdep.ResolvedJSDependency
 import org.scalajs.core.tools.linker.{ModuleInitializer, Semantics, StandardLinker, ModuleKind => ScalaJSModuleKind}
 import org.scalajs.core.tools.logging.ScalaConsoleLogger
 import org.scalajs.jsenv._
 import org.scalajs.testadapter.TestAdapter
-import mill.scalajslib.api.{JsEnvConfig, ModuleKind}
+
 class ScalaJSWorkerImpl extends mill.scalajslib.api.ScalaJSWorkerApi {
+    
+  var scalaJSModuleKind: ScalaJSModuleKind = null;
```
Comment (lihaoyi) [inline scalajslib/worker/0.6/src/ScalaJSWorkerImpl.scala]: @javaduke we should not have mutable variables if we can avoid it. If we need this field in `getFramework`, then we should pass it in

**Comment by lihaoyi** [comment]
one more comment. Just ignore appveyor for now

File: scalajslib/worker/0.6/src/ScalaJSWorkerImpl.scala (hunk 1 above)
Comment (javaduke) [reply scalajslib/worker/0.6/src/ScalaJSWorkerImpl.scala]: Good point, fixed.

**Pull request merged**